ethpm/utils/uri.py

```diff
@@ -90,3 +91,15 @@ def validate_blob_uri_contents(contents: bytes, blob_uri: str) -> None:
         raise ValidationError(
             f"Hash of contents fetched from {blob_uri} do not match its hash: {blob_hash}."
         )
+
+
+def parse_registry_uri(uri: str) -> Tuple[str, ...]:
+    """
+    Validate and return (authority, pkg name, version) from a valid registry URI
+    """
+    validate_registry_uri(uri)
+    parsed_uri = parse.urlparse(uri)
+    authority = parsed_uri.netloc
+    pkg_name = parsed_uri.path.strip("/")
+    pkg_version = parsed_uri.query.lstrip("version=").strip("/")
+    return authority, pkg_name, pkg_version
```

Comment:
Maybe use a `NamedTuple` for the return type here?  That's probably ideal so that it's easy for consumers of this API to keep this info packaged up together while maintaining strong type safety and having property access to the individual components.  (make sure the class is available in the public namespace so that consumers of this package can use it in their type hints)